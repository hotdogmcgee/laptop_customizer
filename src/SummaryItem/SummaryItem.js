import React from 'react'

class ShoppingItem extends React.Component {
    render() {

        const { name, label, cost } = this.props
        return (
            <div className="summary__option" label={label}>
                <div className="summary__option__label">{label}  </div>
                <div className="summary__option__value">{name}</div>
                <div className="summary__option__cost">
                  { new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD'})
                  .format(cost) }
            </div>
        </div>
        )
    }
}

export default ShoppingItem

