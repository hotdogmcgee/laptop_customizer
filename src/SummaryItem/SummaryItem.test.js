import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SummaryItem from './SummaryItem.js'


describe('SummaryItem component', () => {
  it('renders without crashing', () => {

    const div = document.createElement('div');
    ReactDOM.render(<SummaryItem />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<SummaryItem label={"type of thing"} name={"test item"} cost={430}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});