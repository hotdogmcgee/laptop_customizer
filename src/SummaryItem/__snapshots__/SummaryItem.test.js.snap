// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryItem component renders the UI as expected 1`] = `
<div
  className="summary__option"
  label="type of thing"
>
  <div
    className="summary__option__label"
  >
    type of thing
      
  </div>
  <div
    className="summary__option__value"
  >
    test item
  </div>
  <div
    className="summary__option__cost"
  >
    $430.00
  </div>
</div>
`;
